content/en/blog/2019/monitoring-external-service-traffic/index.md

```diff
@@ -0,0 +1,360 @@
+        "destination_app": "unknown",
+        "destination_principal": "unknown",
+        "destination_service": "httpbin.org",
+        "destination_service_name": "PassthroughCluster",
+        "destination_service_namespace": "unknown",
+        "destination_version": "unknown",
+        "destination_workload": "unknown",
+        "destination_workload_namespace": "unknown",
+        "instance": "100.96.2.183:42422",
+        "job": "istio-mesh",
+        "permissive_response_code": "none",
+        "permissive_response_policyid": "none",
+        "reporter": "source",
+        "request_protocol": "http",
+        "response_code": "200",
+        "response_flags": "-",
+        "source_app": "sleep",
+        "source_principal": "unknown",
+        "source_version": "unknown",
+        "source_workload": "sleep",
+        "source_workload_namespace": "default"
+      },
+      "value": [
+        1567033080.282,
+        "1"
+      ]
+    }
+  {{< /text >}}
+
+  Note that the `destination_service_name` label is set to `PassthroughCluster` to
+  indicate that this cluster was hit and the `destination_service` is set to the
+  host of the external service.
+
+* TCP proxy virtual listener - If the external service port doesn't map to any
+  HTTP based service ports within the cluster, this listener is invoked and
+  `istio_tcp_connections_closed_total` is the metric that will be increased:
+
+  {{< text json >}}
+    {
+      "status": "success",
+      "data": {
+        "resultType": "vector",
+        "result": [
+          {
+            "metric": {
+              "__name__": "istio_tcp_connections_closed_total",
+              "connection_security_policy": "unknown",
+              "destination_app": "unknown",
+              "destination_ip": "52.22.188.80",
+              "destination_principal": "unknown",
+              "destination_service": "unknown",
+              "destination_service_name": "PassthroughCluster",
+              "destination_service_namespace": "unknown",
+              "destination_version": "unknown",
+              "destination_workload": "unknown",
+              "destination_workload_namespace": "unknown",
+              "instance": "100.96.2.183:42422",
+              "job": "istio-mesh",
+              "reporter": "source",
+              "response_flags": "-",
+              "source_app": "sleep",
+              "source_principal": "unknown",
+              "source_version": "unknown",
+              "source_workload": "sleep",
+              "source_workload_namespace": "default"
+            },
+            "value": [
+              1567033761.879,
+              "1"
+            ]
+          }
+        ]
+      }
+    }
+  {{< /text >}}
+
+  In this case, `destination_service_name` is set to `PassthroughCluster` and
+  the `destination_ip` is set to the IP address of the external service.
+  The `destination_ip` label can be used to do a reverse DNS lookup and
```

Comment:
aside: this makes me wonder if we should provide utilities for trying to do this very thing.

Reply:
yes, that's a best effort though as IP mappings can always change.